CONTRIBUTING.md

```diff
@@ -25,49 +25,46 @@ all your interactions with the project members and users.
-   Integration (Travis) to test it (tests can be found in the `tests/`
-   directory).
-6. The project's default copyright and header have been included in any new
-   source files.
-7. Make sure you have implemented a local `make test` and all tests succeed
+5. For any new functionality, please write appropriate go tests that will run
+   as part of the Continuous Integration (Circle CI and Travis) systems.
+6. Make sure that the project's default copyright and header have been included 
+   in any new source files.
+7. Make sure you have locally tested using `make test` and that all tests succeed
    before submitting the PR.
-8. Is the code human understandable? This can be accomplished via a clear code
+8. To conform to the Golang standards and idioms, make sure you have done the 
+   following:
+    - Ran `go fmt ./...` to format all `.go` files. We use `go1.11`'s 
+      formatting as our standard
+    - Left a function comment on **every** new exported function and package that
+      your PR has introduced. To learn about how to properly comment Golang code, read [this post on golang.org](https://golang.org/doc/effective_go.html?#commentary)
+9. Is the code human understandable? This can be accomplished via a clear code
    style as well as documentation and/or comments.
-9. The pull request will be reviewed by others, and the final merge must be
-   done by the Singularity project lead, @gmkurtzer (or approved by him).
-10. Documentation must be provided if necessary (next section)
-
-### Documentation
-1. If you are changing any of the following:
-
-   - renamed commands
-   - deprecated / removed commands
-   - changed defaults
-   - backward incompatible changes (recipe file format? image file format?)
-   - migration guidance (how to convert images?)
-   - changed behaviour (recipe sections work differently)
-
-You are **required** to document it in the [changelog](CHANGELOG.md) for the next release.  
-You are also required to provide documentation or a direct pull request to
-the (upcoming) version of the [user docs](https://www.github.io/sylabs/singularity-userdocs) or [admin docs](https://www.github.io/sylabs/singularity-admindocs). Ask for help if you aren't sure where your contribution
-should go.
-2. If necessary, update the README.md, and check the `*.help` scripts under
-   [libexec/cli](libexec/cli) that provide the command line helper output. If
-   you make changes to the internal Python API, make sure to check those
-   changes into the [libexec/python/README.md](libexec/python/README.md) as
-   well.
-
+10. The pull request will be reviewed by others, and finally merged when all
+   requirements are met.
+11. The `CHANGELOG.md` must be updated for any of the following changes:
+    - Renamed commands
+    - Deprecated / removed commands
+    - Changed defaults / behaviors
+    - Backwards incompatible changes
+    - Migration guidance
+    - New features / functionalities (*e.g. a new flag for a command*)
```

Comment:
I liked having bug fixes in the CHANGELOG.md as well, but I'll leave that up to whomever.

Reply:
The reason we don't want bug fixes included in the CHANGELOG is because we want to make an effort to reign in the scope of the CHANGELOG. We want the CHANGELOG to be mostly about functionality, *i.e. intended behavior*, of the product. I think it's okay to leave bug fixes as commit messages

Reply:
So, reading through the old changelog and this is the type of message we want to avoid:

`Fix an error in malloc allocation #1620`

This is something that should be mentioned in a commit message rather than a changelog (this was listed in the old CHANGELOG under a Bug Fixes header). There might be some bug fixes (where there is really broken functionality) that necessitates an entry into the CHANGELOG but I think for most cases (like the example above) it's not useful